lib/FrontendTool/ReferenceDependencies.cpp

```diff
@@ -343,7 +343,7 @@ void ProvidesEmitter::emitExtensionDecl(const ExtensionDecl *const ED,
   auto *NTD = ED->getExtendedNominal();
   if (!NTD)
     return;
-  if (NTD->hasAccess() && NTD->getFormalAccess() <= AccessLevel::FilePrivate) {
+  if (NTD->getFormalAccess() <= AccessLevel::FilePrivate) {
```

Comment:
IIRC we do try to emit swiftdeps files for invalid code, though maybe we shouldn't.